### PR TITLE
[fix] Id generation when using Executor as Modal View

### PR DESCRIPTION
**Pull request by ibelar**
Remove special characters, like dot, from table name when generating View::id.

### CODE REVIEW

**Comment by PhilippGrashoff** [comment]
am curious: why is the table name needed at all?

**Comment by ibelar** [comment]
> am curious: why is the table name needed at all?

If I remember correctly, it was because of callback issues. 
Modal view are generated outside their parent and since Executor are using modal and related on their name or id to carry the model id being used by the action, it was causing issue when executor was create from a reload.
That why we needed a way to make sure Modal name (id) was the same for each Model user action, with no regards to their location in the render tree.

**Comment by mvorisek** [comment]
seems not abstract enough, how name is build normally, is there a method for it?

File: src/UserAction/ConfirmationExecutor.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function afterActionInit(Model\UserAction $action)
 
         $table_name = is_array($action->getModel()->table) ? $getTableName($action->getModel()->table) : $action->getModel()->table;
 
-        $this->id = mb_strtolower($this->name . '_' . $table_name . '_' . $action->short_name);
+        $this->id = preg_replace('~[^A-Za-z0-9\-]~', '_', mb_strtolower($this->name . '_' . $table_name . '_' . $action->short_name));
```
Comment (mvorisek) [inline src/UserAction/ConfirmationExecutor.php:71]: `extends Modal implements JsExecutorInterface` ... the same base

can we dedup these classes a little bit?

**Comment by mvorisek** [comment]
So executor with a non unique name is put to some non-unique render tree?

This should have a Behat test.

File: src/UserAction/ConfirmationExecutor.php [hunk 2]
```diff
@@ -61,17 +61,6 @@ protected function init(): void
      */
     public function afterActionInit(Model\UserAction $action)
     {
-        $getTableName = function ($arr) {
```
Comment (mvorisek) [inline src/UserAction/ConfirmationExecutor.php:64]: why it was needed/is not needed now?

Comment (ibelar) [reply src/UserAction/ConfirmationExecutor.php:64]: Was related to previous callbacks issue with reloading and the fact that the executor was added to $app->html.
Should not be needed now.

Comment (mvorisek) [reply src/UserAction/ConfirmationExecutor.php:64]: perfect! 👍

**Review by mvorisek — APPROVED** [review]

**Pull request merged**